backend/pages/utils/advanced_processing.py

```
@@ -0,0 +1,221 @@
+
+    # Merges last two lines so that we have a single column
+    merged = last_two.aggregate(lambda x: x.dropna().iloc[0] if x.dtype == object else x.sum())
+
+    parsed_df = df.iloc[:-2]
+
+    parsed_df = pd.concat([parsed_df, pd.DataFrame([merged])])
+    
+    return df
+
+def process_competitor_city(df):
+    parsed_df = df.copy();
```

Ruff (0.8.2)

Statement ends with an unnecessary semicolon

Remove unnecessary semicolon

(E703)